[cmake] Enable internal build of libmicrohttpd

fuzzard opened the pull request:
## Description
Enable building internal libmicrohttpd for all platforms

## Motivation and context
consolidate more library building for all platforms

## How has this been tested?
Jenkins

## What is the effect on users?
N/A

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Commit:
[tools/depends][target] Bump libmicrohttpd 1.0.1

Commit:
[tools/depends][target] remove building of libmicrohttpd

Commit:
[win] remove prebuilt libmicrohttpd

Commit:
[cmake] enable build libmicrohttpd internally

fuzzard commented:
Cheers for looking. Should be fixed. i saw the postfix in the patch, but didnt pay attention to the actual characters.

it did raise an interesting thought that i hadnt had before. With the cmakelist patches for windows that are in libs, i should pass through to the build job what the debug postfix is, rather than have it explciitly set in both the find module and the patched cmakelists file. A task for another day though.

thexai reviewed (approved):
Tested Windows x64 and UWP-64

thexai commented:
Note that fails in OSX-ARM64:

```
[  0%] Performing build step for 'build-libmicrohttpd'
/Users/Shared/jenkins/workspace/OSX-ARM64/build/build/build-libmicrohttpd/src/build-libmicrohttpd/build-aux/missing: line 81: autoheader: command not found
WARNING: 'autoheader' is missing on your system.
         You should only need it if you modified 'acconfig.h' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'autoheader' program is part of the GNU Autoconf package:
         <https://www.gnu.org/software/autoconf/>
         It also requires GNU m4 and Perl in order to run:
         <https://www.gnu.org/software/m4/>
         <https://www.perl.org/>
make[3]: *** [MHD_config.h.in] Error 127
make[2]: *** [build/build-libmicrohttpd/src/build-libmicrohttpd-stamp/build-libmicrohttpd-build] Error 2
make[1]: *** [CMakeFiles/build-libmicrohttpd.dir/all] Error 2
make: *** [all] Error 2
```

Commit:
[cmake] Modulehelpers set DEP_BUILDENV for BUILD_COMMAND in BUILD_DEP_TARGET

CONFIGURE_COMMAND and BUILD_COMMAND are separate environments, so set DEP_BUILDENV
to populate tools/info to the build step as well as the configure step

sundermann commented on cmake/modules/FindMicroHttpd.cmake:
```diff
@@ -11,24 +11,84 @@ if(NOT TARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME})
   SEARCH_EXISTING_PACKAGES()
 
+  if(("${${${CMAKE_FIND_PACKAGE_NAME}_SEARCH_NAME}_VERSION}" VERSION_LESS ${${${CMAKE_FIND_PACKAGE_NAME}_MODULE}_VER} AND ENABLE_INTERNAL_LIBMICROHTTPD) OR
+     ((CORE_SYSTEM_NAME STREQUAL linux OR CORE_SYSTEM_NAME STREQUAL freebsd) AND ENABLE_INTERNAL_LIBMICROHTTPD))
```
I have noticed that this also triggers a rebuild on webOS because `CORE_SYSTEM_NAME` is linux on webOS and `ENABLE_INTERNAL_LIBMICROHTTPD` is true so the 2nd condition applies whilst it should only be the first one. Should the 2nd condition also check if it is not a `KODI_DEPENDSBUILD`?
